src/server/debugcmd.cc

```diff
@@ -430,4 +434,55 @@ void DebugCmd::Watched() {
   (*cntx_)->SendStringArr(watched_keys);
 }
 
+void DebugCmd::TxAnalysis() {
+  atomic_uint32_t queue_len{0};
+  atomic_uint32_t free_len{0};
+
+  using SvLockTable = absl::flat_hash_map<string_view, IntentLock>;
+  vector<SvLockTable> lock_table_arr(shard_set->size());
+
+  auto cb = [&](EngineShard* shard) {
+    ShardId sid = shard->shard_id();
+
+    TxQueue* queue = shard->txq();
+
+    if (queue->Empty())
+      return;
+
+    auto cur = queue->Head();
+    do {
+      auto value = queue->At(cur);
+      Transaction* trx = std::get<Transaction*>(value);
+
+      IntentLock::Mode mode = trx->Mode();
+      KeyLockArgs lock_args = trx->GetLockArgs(sid);
```

Comment:
`GetLockArgs()` returns the lock args for the current command. For a multi tx the current command is only the command that is currently executing.

So I guess a possible flow for multi commands is:
```c++
tx->StartMulti(all_keys); // this is where schedule happens

tx->SwitchMultiCmd();
tx->InitByArgs(ars_with_one_key); // but it was scheduled on a lot of keys

// inside cmd
tx->ScheduleSingleHop(); // thats where its blocked, and thats where you look at it, but its has data only for cmd
```

With LOCK_AHEAD mode, the only place this locking information is stored is inside MultiData. So you need one more helper function - scheduled keys - that'll return this info for multi and plain shard args for regular commands